CIP-0054/README.md

```diff
@@ -0,0 +1,227 @@
+
+In this case we define the “own” keyword to mean the tokens held by the same stake key that currently holds the Smart NFT itself.
+
+For example, to create an evolvable NFT which also supports modifier tokens, the “uses” block would look like this: 
+
+```
+	"uses": {
+		"transactions": "own",
+		"tokens": "own"
+	}
+```
+
+We could also monitor a particular smart contract address, for example if we wanted to see how many tokens were listed for sale on a marketplace. The following example creates an NFT that supports modifier tokens and also monitors the tokens held by a script address:
+
+```
+	"uses": {
+		"tokens": [
+			"own",
+			"addr1wywukn5q6lxsa5uymffh2esuk8s8fel7a0tna63rdntgrysv0f3ms"
+		]
+	}
+```
+
+### The renderer key
+
+The idea behind the renderer key is to reduce code duplication in on-chain Javascript by moving the generative code part of the project into a single asset which is minted once in the policy. Each individual NFT within a project is then just a set of input parameters to the generative script - this totally removes the need to fill the metadata of every mint transaction with encoded HTML and Javascript, as is the case with many on-chain Javascript NFTs now. 
+
+When a Smart NFT is encountered which specifies another asset as the renderer, the site rendering the NFT should look-up the referenced asset and render that - the rendering token will then be responsible for reading the appropriate information via the Javascript API below and changing its appearance and/or behaviour based on that. In its simplest form, the rendering token could simply read the current token fingerprint and use that to seed a random number generator - this would enable a generative project to mint NFTs without even changing anything in the metadata and still have the renderer change its appearance for each one. In practice though, it's probably cooler to put actual traits like "colour scheme" or "movement speed" into the metadata and then have the renderer change its behaviour based on that. 
+
+Via the Javascript API, the rendering token will always receive the properties of the child token which specified it as its renderer. This means if you wish to use a renderer with a token which also evolves based on its own transaction history, you will need to specify both "renderer" and "transactions" keys within the child token, and within the renderer token you do not need to specify these keys. 
+
+For example, to create a Smart NFT which is rendered by another token, and is also evolvable based on its own transaction history, the "uses" key would look like this:
+
+```
+	"uses": {
+		"transactions": "own",
+		"renderer": "asset1frc5wn889lcmj5y943mcn7tl8psk98mc480v3j"
+	}
+```
```

Comment:
I am not sure to understand the rationale for storing the rendering code _on-chain_ :thinking: ? It seems to me like a misuse of a blockchain as a database (which it is not!). If the goal is to enforce that only a certain code is executed for rendering an certain NFT, I'd suggest to instead store a hash of that code and have client applications select the right rendering code based on that. 

If the issue is to keep the client as dumb as possible then, I'd suggest to store the code on a system meant to store files such as IPFS, and only record a link / hash to that file on the Cardano blockchain.